Three/include/CGAL/Three/Three.h

```diff
@@ -48,6 +48,21 @@ class THREE_EXPORT Three{
   static Scene_interface* scene();
   static QObject* connectableScene();
   static Three* messages();
+  
+  /*!
+   * Displays a text preceded by the mention "INFO :".
+   */
+  virtual void information(QString) = 0;
+  /*!
+   * Displays a blue text preceded by the mention "WARNING :".
+   */
+  virtual void warning(QString) = 0;
+  /*!
+   * Displays a red text preceded by the mention "ERROR :".
+   */
+  virtual void error(QString) = 0;
```

Comment:
Now the interface class `Message_interface` should be removed.

Reply:
That would require the modification of all the plugins, and the modification of the API of the polyhedron_demo_plugin_interface.

Reply:
Then I do not understand why you have added those three virtual function to `Three`. `MainWindow` already derives from `Message_interface` and now, for example, `MainWindow` has two virtual function named `information`.

Instead, make `Three` derives from `Message_interface`, and remove the direct derivation to it from `MainWindow`.

Actually, the whole thing about `Three::Three` seems wrong.
1. `MainWindow` derives from `Three`.
2. `MainWindow` has a `Three*` as data member.
3. `Three::Three` was supposed to be a class with only static members.

Reply:
3. : The probleme here is addDockWidget(), which is an enhancement for the original qmainwindow::addDockWidget(), but is not used yet. I think it is ultimately  supposed to replace the one from plugin_helper.

Reply:
I don't think it is possible to make those 3 functions static. They need access to member functions from MainWindow (not QMainWindow).

Reply:
As soon as the `MainWindow` object is supposed to be a singleton, then those functions can be implemented as static ones. That is similar to `qDebug()`.

Reply:
done.